ui/app/js/controllers/user_edit_controller.js

```diff
@@ -128,10 +142,18 @@ function ($scope, $modalInstance, CSRF, Permissions, user, users) {
   $scope.addPermission = function() {
     // $scope.permission.options = $scope.permission.options_as_json;
     $scope.saving = true;
+    $scope.usersaved = true;
```

Comment:
Careful here - the user is not actually saved until the Permissions.addPermission request has successfully completed.